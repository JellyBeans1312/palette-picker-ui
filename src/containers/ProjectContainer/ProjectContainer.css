.project-container {
  background-color: yellow;
  width: 50%;
}