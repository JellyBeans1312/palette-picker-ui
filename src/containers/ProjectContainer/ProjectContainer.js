import React from 'react';
import './ProjectContainer.css';
import Project from '../Project/Project'
import { connect } from 'react-redux'


const ProjectContainer = ({ allProjects, allPalettes }) =>
 {
  const projectList = allProjects.map(project => {
    let correspondingPalettes = allPalettes.filter(palette => {
      return palette.project_id === project.id
    }) 
    return <Project props={{...project}} palettes={correspondingPalettes} />
  })
  return (
    <section>
      {projectList}
    </section>
  )
}

const mapStateToProps = state => ({
  allProjects: state.allProjects,
  allPalettes: state.allPalettes
})

export default connect(mapStateToProps)(ProjectContainer);