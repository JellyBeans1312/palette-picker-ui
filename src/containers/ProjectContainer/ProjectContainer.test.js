import React from 'react';
import { shallow } from 'enzyme';
import { ProjectContainer } from './ProjectContainer';

describe('ProjectContainer', () => {
  let wrapper, mockAllProjects, mockAllPalettes;

  beforeEach(() => {
    mockAllProjects = [
      {id: 51, 
        project_name: "Aidan's fantastic project"
      },
      {
        id: 52,
        project_name: "Eric's great project"
      }
    ]
    mockAllPalettes = [
      {
        id: 21,
        palette_name: "asdf",
        project_id: 50,
        color_one: "#67cd51",
        color_two: "#15fe0d",
        color_three: "#507dbb",
        color_four: "#784994",
        color_five: "#2570dd"
      }
    ]
    wrapper = shallow(<ProjectContainer 
      allProjects={mockAllProjects}
      allPalettes={mockAllPalettes}
    />);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});