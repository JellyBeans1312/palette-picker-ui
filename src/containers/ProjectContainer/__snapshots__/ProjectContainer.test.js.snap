// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectContainer should match the snapshot with all data passed in correctly 1`] = `
<section
  className="project-container"
>
  <Connect(Project)
    key="51"
    palettes={Array []}
    props={
      Object {
        "id": 51,
        "project_name": "Aidan's fantastic project",
      }
    }
  />
  <Connect(Project)
    key="52"
    palettes={Array []}
    props={
      Object {
        "id": 52,
        "project_name": "Eric's great project",
      }
    }
  />
</section>
`;
